Tests/Unit/InstanceTests.swift

```diff
@@ -80,4 +80,162 @@ class InstanceTests: XCTestCase {
     let isDifferent = dcOne !== dcTwo
     XCTAssertTrue(isDifferent)
   }
+
+  struct VariableTypeOne: OperationVariable {
+    var email: String
+    var handle: String
+    var someNumber: Int
+  }
+
+  // same query name, same variables, should return same instance
+  func testSameQueryRefInstance() throws {
+    let dc = DataConnect.dataConnect(connectorConfig: fakeConnectorConfigOne)
+
+    let queryName = "someQuery"
+    let variableOne = VariableTypeOne(email: "aa@g.com", handle: "aa@", someNumber: 12345)
+    let variableTwo = VariableTypeOne(email: "aa@g.com", handle: "aa@", someNumber: 12345)
+
+    let refOne = dc.query(
+      name: queryName,
+      variables: variableOne,
+      resultsDataType: Int.self,
+      publisher: .observableObject
+    ) as! QueryRefObservableObject<Int, VariableTypeOne>
+
+    let refTwo = dc.query(
+      name: queryName,
+      variables: variableTwo,
+      resultsDataType: Int.self,
+      publisher: .observableObject
+    ) as! QueryRefObservableObject<Int, VariableTypeOne>
+
+    let isSameInstance = refOne === refTwo
```

Comment:
Consider changing the last 2 lines of each of these tests to be one line (instead of 2 lines), always using XCTAssertTrue. IMO this improves readability because it reads left to right like a book.

e.g.

```
XCTAssertTrue(refOne === refTwo)
```

```
XCTAssertTrue(refOne !== refTwo)
```

Reply:
This is done.